packages/samples/headless-ssr-commerce/components/context-dropdown.tsx

```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import {useContext, useEngine} from '@/lib/commerce-engine';
+import {
+  CommerceEngine,
+  ContextOptions,
+  loadProductListingActions,
+  loadSearchActions,
+} from '@coveo/headless-react/ssr-commerce';
+
+// A hardcoded list of storefront associations for switching app context by language, country, and currency.
+// Found in the admin console under "Storefront Associations," this list is static for demonstration purposes.
+// In a real application, these values would likely come from sources like environment variables or an API.
+const storefrontAssociations = [
+  'en-CA-CAD',
+  'fr-CA-CAD',
+  'en-GB-GBP',
+  'en-US-USD',
+];
+
+export default function ContextDropdown({
+  useCase,
+}: {
+  useCase?: 'listing' | 'search';
+}) {
+  const {state, controller} = useContext();
+  const engine = useEngine();
+
+  return (
+    <div>
+      <p></p>
+      Context dropdown :
+      <select
+        value={`${state.language}-${state.country}-${state.currency}`}
+        onChange={(e) => {
+          const [language, country, currency] = e.target.value.split('-');
+          controller?.setLanguage(language);
+          controller?.setCountry(country);
+          controller?.setCurrency(currency as ContextOptions['currency']);
+
+          useCase === 'search'
+            ? engine?.dispatch(
+                loadSearchActions(engine as CommerceEngine).executeSearch()
+              )
+            : useCase === 'listing' &&
+              engine?.dispatch(
+                loadProductListingActions(
+                  engine as CommerceEngine
+                ).fetchProductListing()
+              );
+        }}
```

Comment:
The reason I did this this way is because I ran into a problem while using the useEngineHook. I would do something like this

```
export default function ContextDropdown({
  useCase,
}: {
  useCase?: 'listing' | 'search';
}) {
  const {state, controller} = useContext();
  const engine = useEngine();
  const {executeSearch} = loadSearchActions(engine);
  ...
```

this would create errors since the engine is not hydrated at first. 
<img width="908" alt="image" src="https://github.com/user-attachments/assets/446bff31-16b7-449a-b98e-f5ebf7c9464a">

In those particular cases, when working in a random client component, **how would I know when my page is hydrated ?** Do we have some sort of variable or hook we can depend on in a useEffect so it properly loads the actions when the engine is hydrated ? I could not find an answer on how to do this so I put all the code directly in the onChange function.

Reply:
Loading & dispatching actions seems complicated in SSR 🤔 
We should probably either steer people away from those in favor of controllers, or improve the DX.